packages/dev/core/src/FlowGraph/Blocks/Data/flowGraphCachedOperationBlock.ts

```diff
@@ -0,0 +1,35 @@
+import type { IFlowGraphBlockConfiguration } from "../../flowGraphBlock";
+import { FlowGraphBlock } from "../../flowGraphBlock";
+import type { FlowGraphContext } from "../../flowGraphContext";
+import type { FlowGraphDataConnection } from "../../flowGraphDataConnection";
+import type { RichType } from "../../flowGraphRichTypes";
+
+const CACHE_NAME = "cachedOperationValue";
+const EXEC_ID_NAME = "executionId";
+
+/**
+ * @experimental
+ */
+export abstract class FlowGraphCachedOperationBlock<OutputT> extends FlowGraphBlock {
+    public readonly output: FlowGraphDataConnection<OutputT>;
+
+    constructor(outputRichType: RichType<OutputT>, config?: IFlowGraphBlockConfiguration) {
+        super(config);
+
+        this.output = this._registerDataOutput("output", outputRichType);
+    }
+    public abstract _doOperation(context: FlowGraphContext): OutputT;

```

Comment:
I think you have to add a `@internal` tag if the method is not supposed to be visible / used by the end user?